docs/adrs/2023-07-18-customize-listener-pod.md

```diff
@@ -0,0 +1,54 @@
+# Customize listener pod
+
+**Status**: Proposed
+
+## Context
+
+The Autoscaling listener is a critical component of the autoscaling solution, as it monitors the workload and communicates with the ephemeral runner set to adjust the number of runners as needed.
+
+The problem can arise when cluster policies are configured to disallow pods with (or without) certain fields. Since the Autoscaling listener pod is an internal detail that is not currently customizable, it can be a blocker for users with Kubernetes clusters that enforce such policies.
+
+## Decision
+
+Expose field on the `AutoscalingRunnerSetSpec` resource called `ListenerTemplate` of type `PodTemplateSpec`.
+
+Expose field on the `AutoscalingListenerSpec` resource called `Template` of type `PodTemplateSpec`.
+
+The `AutoscalingRunnerSetController` is responsible for creating the `AutoscalingListener` with the `ListenerTemplate`.
+
+The `AutoscalingListenerController` then creates the listener pod based on the default spec, and the customized spec.
+
+List of fields that are going to be ignored by the merge:
+
+- `spec.serviceAccountName`: Created by the AutoscalingListener.
+- reserved `metadata.labels`: Labels that collide with reserved labels used by the system are ignored.
+- reserved `spec.containers[0].env`: Environment variables used by the listener application
+- `metadata.name`: Name of the listener pod
+- `metadata.namespace`: Namespace of the listener pod
```

Comment:
I would also propose not adding ability to change:
`spec.restartPolicy` and `spec.imagePullSecrets`

`imagePullSecrets` is set by the controller, and restart policy allows us to explicitly remove and re-create a listener. I don't see how changing it may benefit someone so I would prefer to ignore it for now :relaxed: 

Any thoughts (@TingluoHuang, @Link-)?
If it easier to just see the code, here is the new PR implementing this ADR: https://github.com/actions/actions-runner-controller/pull/2758

Reply:
We're allowing everything else, we might allow these as well

Reply:
Awesome, I included this setting if specified by the template customization